lightwood/mixers/nn/nn.py

```diff
@@ -777,19 +730,19 @@ def iter_fit(self, ds, initialize=True, subset_id=None):
     nums = [data['x'][i] * data['y'][i] for i in range(10)]
 
     data['z'] = [i + 0.5 for i in range(10)]
+    data['z`'] = ['low' if i < 50 else 'high' for i in nums]
 
     data_frame = pandas.DataFrame(data)
     ds = DataSource(data_frame, config)
     ds.prepare_encoders()
-    predict_input_ds = DataSource(data_frame[['x', 'y']], config)
-    predict_input_ds.prepare_encoders()
-    ####################
 
     mixer = NnMixer({}, config)
 
     for i in mixer.iter_fit(ds):
-        if i < 0.01:
-            break
+        break
+    mixer.fit(ds,ds, stop_training_after_seconds=50)
 
+    predict_input_ds = DataSource(data_frame[['x', 'y']], config)
+    predict_input_ds.prepare_encoders()
```

Comment:
i dont get it, how are you going to prepare encoders after you fit the mixer?

Reply:
That's just on the `predict_input_ds` (the one we use for predicting), the `ds` object (the one that's used when we call `.fit` is being prepared above).

Reply:
ok